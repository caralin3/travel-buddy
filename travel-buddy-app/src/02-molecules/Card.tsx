import React from 'react';
import { Link } from 'react-router-dom';
import { Card as RBCard, CardBody, CardTitle, CardSubtitle, CardText, Button } from 'reactstrap';
import { LinkItem } from '../types';

export interface CardProps {
  buttonLabel?: string;
  icon?: string;
  link?: LinkItem;
  onButtonClick?: VoidFunction;
  subtitle?: string;
  title: string;
}

export const Card: React.FC<CardProps> = ({ buttonLabel, children, icon, link, onButtonClick, subtitle, title }) => (
  <RBCard>
    <CardBody>
      <CardTitle tag="h5">{title}</CardTitle>
      {!!subtitle && (
        <CardSubtitle className="mb-2 text-muted" tag="h6">
          {subtitle}
        </CardSubtitle>
      )}
      <CardText>{children}</CardText>
      {buttonLabel && (
        <Button onClick={onButtonClick} color="primary">
          {buttonLabel}
        </Button>
      )}
      {!!link && (
        <Link className="btn btn-primary" to={link.path}>
          {link.label}
        </Link>
      )}
    </CardBody>
  </RBCard>
);
