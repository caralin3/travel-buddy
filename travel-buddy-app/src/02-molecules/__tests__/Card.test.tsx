import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Card } from '../Card';

describe('Card', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Card title="Card" buttonLabel="Click" link={{ path: '#', label: 'Link' }} subtitle="Subtitle" />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
