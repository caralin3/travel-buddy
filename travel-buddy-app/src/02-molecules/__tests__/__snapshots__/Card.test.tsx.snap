// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<Card
  tag="div"
>
  <CardBody
    tag="div"
  >
    <CardTitle
      tag="h5"
    >
      Card
    </CardTitle>
    <CardSubtitle
      className="mb-2 text-muted"
      tag="h6"
    >
      Subtitle
    </CardSubtitle>
    <CardText
      tag="p"
    />
    <Button
      color="primary"
      tag="button"
    >
      Click
    </Button>
    <Link
      className="btn btn-primary"
      to="#"
    >
      Link
    </Link>
  </CardBody>
</Card>
`;
